main/src/cgeo/geocaching/maps/PositionHistory.java

```diff
@@ -33,32 +33,16 @@ public PositionHistory() {
 
     // save current coords to permanent storage
     private void saveToStorage(final Location coordinates) {
-        AndroidRxUtils.andThenOnUi(Schedulers.io(), new Runnable() {
-            @Override
-            public void run() {
-                DataStore.saveTrailpoint(coordinates);
-            }
-        }, new Runnable() {
-            @Override
-            public void run() {
-                // nothing to do UI-wise
-            }
+        Schedulers.io().scheduleDirect(() -> {
```

Comment:
For all other calls of scheduleDirect, we use:
```suggestion
        Schedulers.io().scheduleDirect(new Runnable() {
```

Reply:
It has been like this (with `new Runnable()`) before. But we do have lambda functions now since we support Java 8, so we can make use of the new syntax available (less & cleaner code, in this case).

Reply:
I think it's more consistent to use `new Runnable()` here and add another PR that changes all occurrences to use lambda.

Reply:
ok, updated accordingly